lib/src/main/java/io/lantern/observablemodel/SharedPreferencesAdapter.kt

```diff
@@ -5,59 +5,61 @@ import java.util.*
 import kotlin.collections.HashMap
 
 /**
- * Allows accessing an ObservableModel using the SharedPreferences API
+ * Allows accessing an ObservableModel using the SharedPreferences API.
+ *
+ * @param model the model in which to store the preferences
+ * @param prefix a prefix to prepend to all preference keys before storing them in the model
+ * @param initialValues the model will be populated with values from this SharedPreferences for any values that haven't already been set (useful for migrating from a regular SharedPreferences)
  */
 internal class SharedPreferencesAdapter(
     val model: ObservableModel,
     val prefix: String,
-    val fallback: SharedPreferences?
-) :
-    SharedPreferences {
+    initialValues: SharedPreferences?
+) : SharedPreferences {
     private val listenerIds = Collections.synchronizedList(ArrayList<ListenerId>())
 
-    override fun getAll(): MutableMap<String, *> {
-        val all = fallback?.let { it.all as MutableMap<String, Any?> } ?: HashMap()
-        model.list<Any>("${prefix}%").forEach { entry ->
-            all[unprefixedPath(entry.path)] = entry.value
+    init {
+        initialValues?.all?.let {
```

Comment:
I like this approach. Keeping the fallback SharedPreferences around all the time just for fallback purpose is not really nice. With this approach, once the initialization is done, we can totally depend on the observable only.